includes/formatting.php

```diff
@@ -57,6 +57,60 @@ function get_body_contents( $content ) {
 add_filter( 'airstory_before_insert_content', __NAMESPACE__ . '\get_body_contents', 1 );
 
 /**
+ * Sideload a single image from a remote URL.
+ *
+ * @param string $url      The remote URL for the image.
+ * @param int    $post_id  Optional. The post the newly-uploaded image should be attached to.
+ *                         Default is 0 (unattached).
+ * @param array  $metadata Optional. Additional post meta keys to assign once the attachment post
+ *                         has been created. These keys and values are assumed to be sanitized.
+ *                         Default is an empty array.
+ */
+function sideload_image( $url, $post_id = 0, $metadata = array() ) {
+	require_once ABSPATH . 'wp-admin/includes/media.php';
+	require_once ABSPATH . 'wp-admin/includes/file.php';
+	require_once ABSPATH . 'wp-admin/includes/image.php';
+
+	$tmp_file   = download_url( esc_url_raw( $url ) );
+	$file_array = array(
+		'name'     => basename( $url ),
+		'tmp_name' => $tmp_file,
+	);
+
+	// Something went wrong downloading the image.
+	if ( is_wp_error( $tmp_file ) ) {
+		@unlink( $file_array['tmp_name'] ); // @codingStandardsIgnoreLine
+		trigger_error( esc_html( $tmp_file->get_error_message() ), E_USER_WARNING );
+
+		return 0;
+	}
+
+	// Sideload the media.
+	$image_id = media_handle_sideload( $file_array, $post_id );
+
+	if ( is_wp_error( $image_id ) ) {
+		@unlink( $file_array['tmp_name'] ); // @codingStandardsIgnoreLine
+		trigger_error( esc_html( $image_id->get_error_message() ), E_USER_WARNING );
+
+		return 0;
+	}
+
+	/*
+	 * Finally, store post meta. We'll always set _airstory_origin (the original image URL), but any
+	 * non-empty values in $metadata will also be set.
+	 */
+	add_post_meta( $image_id, '_airstory_origin', esc_url( $url ) );
+
+	foreach ( (array) $metadata as $meta_key => $meta_value ) {
```

Comment:
Casting it to an array will result in an empty array if it isn't already, which basically causes:

```php
foreach ( array() as $key => $value ) {
  // Nothing here will get run.
}
```

Not a major concern, IMO, but you're not wrong. I've added an `empty()` check around the block.